api/src/main/java/com/netflix/iceberg/types/IndexByLowerCaseName.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg.types;
+
+import java.util.Locale;
+
+/**
+ * Allows search by name in a case insensitive way.
+ */
+public class IndexByLowerCaseName extends IndexByName {
```

Comment:
Why not use `IndexByName` and then copy that index and transform the keys?

Reply:
Yes, that sounds cheaper. Will do.

Reply:
Done.